scripts/module/formapplications/sequencer-database-viewer.js

```diff
@@ -1,179 +1,227 @@
-
-        let filter = html.find('select[name="pack-select"]');
-        let input = html.find('input[name="search-field"]');
-        this.player = html.find('.database-player');
-        this.image = html.find('.database-image');
-
-        let filterDebounce = debounce(() => {
-            this.search = input.val();
-            this.filter = filter.val();
-            this.refreshView(html);
-        }, 500)
-
-        filter.change(filterDebounce);
-        input.keyup(function (e) {
-            filterDebounce()
-        });
-
-        let self = this;
-        html.find('.btn_play').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.playAsset.bind(self)(entry);
-        });
-        html.find('.btn_copy_databasepath').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.copyText($(this), entry, false);
-        });
-        html.find('.btn_copy_filepath').click(function () {
-            let entry = $(this).siblings(".database-entry-text").text();
-            self.copyText($(this), entry, true);
-        });
+  }
+
+  /** @override */
+  static get defaultOptions() {
+    return foundry.utils.mergeObject(super.defaultOptions, {
+      title: "Sequencer Database Viewer",
+      template: `modules/sequencer/templates/sequencer-database-template.html`,
+      classes: ["dialog"],
+      width: 900,
+      height: 425,
+    });
+  }
+
+  getSearchRegex() {
+    return new RegExp(this.search, "gu");
+  }
+
+  getFilteredEntries() {
+    const searchParts = this.search.split("|");
+    const { search, filter } = this;
+    return this.entries.filter((part) => {
+      return (
+        (filter === "all" || filter === part.pack) &&
+        (search === "" ||
+          part.entry.match(this.getSearchRegex())?.length >= searchParts.length)
+      );
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  getData() {
+    const data = super.getData();
+    data.packs = this.packs;
+    return data;
+  }
+
+  /** @override */
+  activateListeners($html) {
+    super.activateListeners($html);
+
+    const [html] = $html;
+
+    const scroller = html.querySelector(".database-entries");
+    const wrapper = html.querySelector(".database-entries-wrapper");
+
+    const rerenderList = () => {
+      const entries = this.getFilteredEntries();
+
+      const wrapperHeight = entries.length * 20;
+
+      const scrolledIndex = Math.floor(scroller.scrollTop / 20) - 5;
+      const startIndex = clamp(
+        scrolledIndex,
+        Math.max(entries.length - MAX_NODES, 0)
+      );
+
+      this.listItems.forEach((el, i) => el.update(entries[startIndex + i]));
+
+      wrapper.style.height = `${wrapperHeight}px`;
+      wrapper.style.paddingTop = startIndex * 20 + "px";
+    };
+
+    rerenderList();
+    scroller.addEventListener("scroll", rerenderList, { passive: true });

```

Comment:
Guide: We could debounce here but it seems to break the illusion quite heavily, we don't seem to get much perf issues without it and `passive: true` means we don't mess around with the natural scrolling